docs/source/sparse.rst

```diff
@@ -7,51 +7,162 @@
-elements being equal to zero which means that a lot of memory as well
-as processor resources can be spared if only the non-zero elements are
-stored or/and processed. Various sparse storage formats (`such as COO,
-CSR/CSC, LIL, etc.`__) have been developed that are optimized for a
-particular structure of non-zero elements in sparse arrays as well as
-for specific operations on the arrays. PyTorch supports the following
-sparse storage formats: :ref:`COO<sparse-coo-docs>`,
-:ref:`CSR<sparse-csr-docs>`, :ref:`CSC<sparse-csc-docs>`,
-:ref:`BSR<sparse-bsr-docs>`, and :ref:`BSC<sparse-bsc-docs>`.
-
-__ https://en.wikipedia.org/wiki/Sparse_matrix
+.. warning::
 
-.. note::
+  The PyTorch API of sparse tensors is in beta and may change in the near future.
+  We highly welcome feature requests, bug reports and general suggestions as Github issues.
+
+Why and when to use sparsity
+++++++++++++++++++++++++++++
+
+By default PyTorch stores :class:`torch.Tensor` stores elements contiguously
+physical memory. This leads to efficient implementations of various array
+processing algorithms that require fast access to elements.
+
+Now, some users might decide to represent data such as graph adjacency
+matrices, pruned weights or points clouds by Tensors whose *elements are
+mostly zero valued*. We recognize these are important applications and aim
+to provide performance optimizations for these use cases via sparse storage formats.
+
+Various sparse storage formats such as COO, CSR/CSC, LIL, etc.
+have been developed over the years. In a nutshell, they all compress data by
+tracking the positions of non-zero valued elements and perhaps some,
+but *most importantly* not all, zero valued elements. We call the uncompressed
+values *specified*. They *may* be zero. In contrast *unspecified* elements
+*must* be zero.
+
+By compressing repeat zeros sparse storage formats aim to save memory
+and computational resources on various CPUs and GPUs. Especially for high
+degrees of sparsity or highly structured sparsity this can have significant
+performance implications. As such sparse storage formats can be seen as a
+performance optimization.
+
+Like many other performance optimization sparse storage formats are not
+always advantageous. Details such as the distribution of zero valued
+elements, the level of sparsity and the user's device matter.
+
+We aim to make it easy to try these different compression techniques,
+but we do not yet provide a programmatic means of determining the best one
+for your particular application. Of course you are always welcome to
+ask for help in our discussion forum https://dev-discuss.pytorch.org/.
+
+Functionality overview
+++++++++++++++++++++++
+
+We want it to be straightforward to construct a sparse Tensor from a
+given dense Tensor by providing conversion routines for each layout.
+
+In the next example we convert a 2D Tensor with default strided layout
+to a 2D Tensor backed by the COO memory layout. Only values and indices
```

Comment:
is this the only sparse page? It might be useful to briefly describe the formats in a section before this so users know the acronyms you're referring nto.

Layout overview
+++++

We briefly detail the formats we currently support. 

Block Sparse Row compression (BSR) ...
Coordinate Format (COO) ...


Functionality Overview
++++++
```

Reply:
Coverage is detailed in the next section. I expect users who are interested in this to be generally familiar with these common layouts.